randovania/games/fusion/exporter/patch_data_factory.py

```diff
@@ -242,23 +243,57 @@ def _create_nav_text(self) -> dict:
         hints = {}
         restricted_hint = ""
         operations_hint = ""
-        hint_counter = 0
+        artifact_locations = guaranteed_item_hint.find_locations_that_gives_items(
+            artifacts, self.description.all_patches, self.players_config.player_index
+        )
+        artifact_pickup_indexes = {}
+        for resource, locations in artifact_locations.items():
+            for location in locations:
+                player, location_info = location
+                artifact_pickup_indexes.update({location_info.location: resource.long_name})
+
+        fusion_bosses = [
+            "Arachnus",
+            "Yakuza",
+            "Ridley",
+            "ChargeCoreX",
+            "Zazabi",
+            "Nettori",
+            "WideCoreX",
+            "Serris",
+            "Nightmare",
+            "MegaX",
+            "WaveCoreX",
+        ]
+        fusion_boss_pickup_indexes = []
+        for pickup_node in self.game.region_list.iterate_nodes_of_type(PickupNode):
+            if "source" in pickup_node.extra:
+                for boss in fusion_bosses:
+                    if boss in pickup_node.extra["source"]:
+                        fusion_boss_pickup_indexes.append(pickup_node.pickup_index)
+
+        artifacts_on_bosses = []
+        for boss_index in fusion_boss_pickup_indexes:
+            if boss_index in artifact_pickup_indexes:
+                artifacts_on_bosses.append(artifact_pickup_indexes[boss_index])
+
         if metroid_precision != SpecificPickupHintMode.DISABLED:
-            # loop through all metroids and place them alternating on Restricted then Operations Nav Rooms
+            # loop through all metroids and place ones on bosses on Operations Deck and the rest on Restricted Area
             for _, text in metroid_hint_mapping.items():
                 if "has no need to be located" in text:
                     continue
-                if hint_counter % 2 == 0:
-                    restricted_hint = restricted_hint + text + " "
-                else:
+                if _.long_name in artifacts_on_bosses:
```

Comment:
A variable called `_` implies "this variable is unused, but is declared for one reason or another".
If you actually use the variable like here, then you need to rename it.

(Altho for clarification this variable shouldnt have existed earlier and the code shouldve just done `for text in mapping.values()`)